store.go

```diff
@@ -1,199 +1,207 @@
 package kallax
 
 import (
+	"database/sql"
 	"errors"
+	"fmt"
 
-	"gopkg.in/mgo.v2"
-	"gopkg.in/mgo.v2/bson"
+	"github.com/Masterminds/squirrel"
+	"github.com/lann/builder"
 )
 
 var (
 	// ErrNonNewDocument non-new documents cannot be inserted
-	ErrNonNewDocument = errors.New("Cannot insert a non new document.")
+	ErrNonNewDocument = errors.New("cannot insert a non new document")
 	// ErrNewDocument a new documents cannot be updated
-	ErrNewDocument = errors.New("Cannot updated a new document.")
-	// ErrEmptyQueryInRaw an empty query cannot be used on any *Raw method
-	ErrEmptyQueryInRaw = errors.New("Empty queries are not allowed on raw ops.")
-	// ErrEmptyID a document without Id cannot be used with Save method
-	ErrEmptyID = errors.New("A document without id is not allowed.")
+	ErrNewDocument = errors.New("cannot updated a new document")
+	// ErrEmptyID a document without ID cannot be used with Save method
+	ErrEmptyID = errors.New("a record without id is not allowed")
 )
 
+// Store is a structure capable of retrieving records from a concrete table in
+// the database.
 type Store struct {
-	db         *mgo.Database
-	collection string
+	db       squirrel.DBProxy
+	schema   Schema
+	inserter squirrel.InsertBuilder
+	updater  squirrel.UpdateBuilder
+	deleter  squirrel.DeleteBuilder
 }
 
-// NewStore returns a new Store instance
-func NewStore(db *mgo.Database, collection string) *Store {
+// NewStore returns a new Store instance.
+func NewStore(db *sql.DB, schema Schema) *Store {
+	builder := squirrel.StatementBuilder.PlaceholderFormat(squirrel.Dollar)
+	proxy := squirrel.NewStmtCacher(db)
 	return &Store{
-		db:         db,
-		collection: collection,
+		db:       proxy,
+		schema:   schema,
+		inserter: builder.Insert(schema.Table()).RunWith(proxy),
+		updater:  builder.Update(schema.Table()).RunWith(proxy),
+		deleter:  builder.Delete(schema.Table()).RunWith(proxy),
 	}
 }
 
-// Insert insert the given document in the collection, returns error if no-new
-// document is given. The document id is setted if is empty.
-func (s *Store) Insert(doc DocumentBase) error {
-	if !doc.IsNew() {
+// Insert insert the given record in the table, returns error if no-new
+// record is given. The record id is set if it's empty.
+func (s *Store) Insert(record Record) error {
+	if !record.IsPersisted() {
```

Comment:
isNew != isPersisted
then... it should be 
`if record.IsPersisted() {
    return ErrNonNewDocument
}`